### ISSUE TITLE
ConnectionsDetachable does not allow dragging multiple connections from endpoints when false

### ISSUE THREAD

**Issue by lizardoluis** [jsplumb/jsplumb#582]
When setting `ConnectionsDetachable` as `false` in the default setting of an instance, it does not allow to drag more than one connection from an endpoint.

By the definition of this variable, it seems to be a defect.

> **ConnectionsDetachable** - Whether or not Connections are detachable by default using the mouse.

To reproduce this situation, just add `ConnectionsDetachable : false` to the instance default setting in the Drag and Drop Demo. Then, try to drag two connections **from** the same green dot endpoint.

**Comment by brian-davies** [comment]
I'm having this issue as well.  I can add a little more information:  This is a result of a change in version 1.7.3.

In 1.7.2 and earlier, ConnectionsDetachable: false would prevent detaching without interfering with dragging a new connection in all cases.  In 1.7.3 and later, ConnectionsDetachable: false allows dragging one new connection from a source, but considers a click on a source with an existing connection as an attempt to detach the existing connection, and so blocks it entirely, rather than passing the click through to allow a new connection to be dragged.

I agree -- as per the docs, this seems like a defect.

To get the desired behavior, allowing dragging new connections until MaxConnections without allowing detaching, I'm stuck on version 1.7.2.

The best I've come up if I want to upgrade is to have ConnectionsDetachable: true, and then block completing a detach like such:

	jsPlumb.bind ("beforeDetach", function (connection)
	{
		return false;
	});

That is a bit ugly -- dragging from a source works perfectly (whether there are connections on that endpoint or not), but dragging from a target with an attached connection allows users to start the detach, even though it's going to thwart that detach in all cases.

**Comment by brian-davies** [comment]
Looks like 1.7.6 includes beforeStartDetach, which might get rid of the ugliness.  Still seems like a defect, that ConnectionsDetachable doesn't work as advertised.  But there might be a better workaround out there.

**Comment by sporritt** [comment]
although i'd agree there's a bug here - and forgive me for splitting hairs - it is not the definition of ConnectionsDetachable that indicates the bug. ConnectionsDetachable works exactly as advertised and documented: it prevents connections from being detached.

There is a stopgap measure you can put in place until this issue gets fixed. When drag starts on an endpoint jsPlumb requests a connection from a function on the endpoint called `connectorSelector`. It cannot be passed in to the `addEndpoint` method, nor specified on an endpoint definition, but it can be set on an endpoint after it is created, for example:

```
var e1 = instance.addEndpoint('dragDropWindow1', { anchor: [0.5, 1, 0, 1] }, exampleEndpoint2);
e1.connectorSelector = function() { return null; };
```
of course you'd want to wrap the `addEndpoint` method to do this always, not one at a time like I have it here.

This tricks jsPlumb into allowing the connection drag start because it doesnt get back a non-detachable connection like it would otherwise - the default implementation is this:

```
this.connectorSelector = function() { return this.connections[0]; }
```

This does not circumvent the `maxConnections` test.

**Comment by brian-davies** [comment]
Ah, split all the hairs you'd like!  Many thanks for confirming this is a bug, and for the workaround!

**Comment by sporritt** [comment]
i've just released 2.7.15 with a fix for this..upon discovering that the connection it was given is not detachable, if the endpoint is not full it sets the connection reference to null and then proceeds along the path of adding a new one.
